### PR TITLE
Added moving powerups and refactored collision detection

### PR DESCRIPTION
**Pull request by Jelly-Yu**
@yihuaqi

### CODE REVIEW

File: app/src/main/java/hacking/to/the/gate/GameManager.java [hunk 1]
```diff
@@ -99,39 +99,93 @@ public void setSelfJetDest(MotionEvent event){
...
+        return Math.pow(
+                c1.getPosition().getPositionX()-c2.getPosition().getPositionX(),
+                2)
+                +Math.pow(
+                c1.getPosition().getPositionY()-c2.getPosition().getPositionY(),
+                2)
+                < Math.pow(c1.getRadius()+c2.getRadius(),2);
+
+    }
+
```
Comment (yihuaqi) [inline app/src/main/java/hacking/to/the/gate/GameManager.java]: Should make this a method of CircleCollider class.
GameManager shouldn't know the detail of how the collision is detected.

File: app/src/main/java/hacking/to/the/gate/PowerUp.java [hunk 2]
```diff
@@ -75,4 +78,19 @@ public float getRadius(){
     public void setVisible(boolean isVisible){
         mIsVisible = isVisible;
     }
+
+    public void movingRandomly(){
+        changeVelocity();
+        mCurPosition = new Position(mCurPosition.getPositionX()+mVelocity.getVelocityX(),
+                mCurPosition.getPositionY()+mVelocity.getVelocityY());
+        super.setPosition(mCurPosition);
+    }
```
Comment (yihuaqi) [inline app/src/main/java/hacking/to/the/gate/PowerUp.java:88]: should stick to the naming convention of tick()

File: app/src/main/java/hacking/to/the/gate/CircleCollider.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package hacking.to.the.gate;
+
+/**
+ * Created by Ruiqian on 8/31/2015.
+ */
```
Comment (yihuaqi) [inline app/src/main/java/hacking/to/the/gate/CircleCollider.java:5]: I don't quite feel it's a good idea to let jet, powerup and bullets to extends this class. When there is no obvious reason, you should prefer composition than inheritance. And in this case, the parent class is just used as a data object.

File: app/src/main/java/hacking/to/the/gate/GameManager.java (hunk 1 above)
Comment (Jelly-Yu) [inline app/src/main/java/hacking/to/the/gate/GameManager.java]: Could you please give me more details or stronger reasons?Actually, I don't think CircleCollider class should have this method even if it's not proper to put the method in GameManager,

**Pull request merged**